Create encode.md for Python Strings .encode()

reganmarie opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

- Closes #3272 
- Created encode.md on docs/content/python/concepts/strings/terms/encode/encode.md
- Added Title, Description, Subjects, Tags, and Catalog Content
- Added description, syntax, example, and codebyte example
- Ran through 3 online plagiarism checkers to ensure 100% unique wording and description, passed with 100%
- Followed content and style guidelines -Checked for spelling and grammar errors
- Read over initial draft to correct for clarity and readability

### Type of Change

- Adding a new entry

### Checklist

<!-- Please check ALL the boxes: -->

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

caupolicandiaz commented on content/python/concepts/strings/terms/encode/encode.md:
```diff
@@ -0,0 +1,67 @@
+
+## Example
+
+In the following example, the `.encode()` method is called twice. The first call does not specify the optional parameters, so the string is encoded in UTF-8 and will throw a `UnicodeDecodeError` if there is failure. The second call specifies that the string will be encoded using `ASCII` and that errors will be handled using `xmlcharrefreplace`.
```
```suggestion
In the following example, the `.encode()` method is called twice. The first call does not specify the optional parameters, so the string is encoded in UTF-8 and will throw a `UnicodeDecodeError` if there is failure. The second call specifies that the string will be encoded using `ascii` and that errors will be handled using `xmlcharrefreplace`.
```

caupolicandiaz commented on content/python/concepts/strings/terms/encode/encode.md:
```diff
@@ -0,0 +1,67 @@
+utf_16_example = example_b.encode(encoding='utf-16', errors='ignore')
+
+print("Default example:", default_example)
+print("ASCII example:", ascii_example)
```
```suggestion
print("UTF-16 example:", utf_16_example)
```

caupolicandiaz reviewed (changes requested):
@reganmarie I've completed my initial review. Please update the entry, and tag me when you've finished making the changes. Thanks

reganmarie commented:
@caupolicandiaz Made the following requested changes! 
- Changed line 29 to the `\uNNNN` characters
- Changed line 30 to the `\N{...}` characters
- Changed line 34 to ascii rather than ASCII
- Changed line 66 to ("UTF-16 example:", utf_16_example)

caupolicandiaz reviewed (approved):
@reganmarie This is ready for a second review.

reganmarie commented:
Thank you @caupolicandiaz! 😊

SSwiniarski reviewed (approved):
@reganmarie Fixed one typo and it looks good to merge.

reganmarie commented:
Thank you @SSwiniarski! 😊
